[5.5.x] Fix sequence overflow bug

bernardjkim opened the pull request:
Marshalling an icmp message into an echo request converts the sequence number into uint16. This causes nethealth to think that the received echo responses are out of order once the echoCounter reaches 65535.

- Convert sequence number and echoCounter to uint16 when comparing.
- Log only echo counter for consistency even though it may not reflect
the actual sequence number used in the icmp echo request.

bernardjkim committed:
Fix sequence overflow bug

- Convert sequence number and echo counter to uint16 when comparing.
- Log only echo counter for consistency even though it may not reflect
the actual sequence number used in the icmp echo request.

bernardjkim committed:
Log expected and received sequence numbers

a-palchikov commented on lib/nethealth/nethealth.go:
```diff
@@ -496,8 +496,9 @@ func (s *Server) processAck(e messageWrapper) error {
 		}
-		if pkt.Seq != peer.echoCounter {
-			return trace.BadParameter("Response sequence doesn't match latest request.")
+		if uint16(pkt.Seq) != uint16(peer.echoCounter) {
```
nit: Could you also remove the type assertion from the `default` case above?
```go
default:
	// unexpected / unknown
	return trace.BadParameter("received unexpected icmp message type").AddField("type", e.message.Type)  // <- HERE
}
```

knisbet commented on lib/nethealth/nethealth.go:
```diff
@@ -496,8 +496,9 @@ func (s *Server) processAck(e messageWrapper) error {
-		if pkt.Seq != peer.echoCounter {
-			return trace.BadParameter("Response sequence doesn't match latest request.")
+		if uint16(pkt.Seq) != uint16(peer.echoCounter) {
+			return trace.BadParameter("response sequence doesn't match latest request. Expected %d, received %d",
```
Probably should add the variables as fields to the error instead of formatting as a string.

Something along the lines of:
```
return trace.BadParameter("response sequence doesn't match latest request.").AddFields(map[string]interface{}{
    "expected": uint16(peer.echoCounter),
    "received": uint16(peer.echoRequest),
}
```

r0mant commented on lib/nethealth/nethealth.go:
```diff
@@ -508,7 +509,7 @@ func (s *Server) processAck(e messageWrapper) error {
 			"peer_name": peer.name,
 			"peer_addr": peer.addr,
-			"id":        pkt.Seq,
+			"id":        peer.echoCounter,
```
Should this be converted to uint16 as well? Otherwise what's in the logs won't match what's actually being compared. We can also include the overall "real" counter e.g.

```go
  "counter": peer.echoCounter,
  "seq": uint16(peer.echoCounter),
```

bernardjkim committed:
Address PR feedback

- Remove unnecessary type assertion
- Log variables as fields
- log both counter and seq
